### ISSUE TITLE
Minified rules in wrong order

### ISSUE THREAD

**Issue by fergaldoyle** [jakubpawlowicz/clean-css#708]
Version: 3.4.8

I noticed this happening with Foundation 6. With the below sample snippet, `ul` and `ol` initially gets `margin: 0` then later `margin-left: 1.25rem;`

After minifying, `margin-left: 1.25rem;` is applied first, then cascaded by `margin: 0`

Command
```
node_modules\.bin\cleancss sample.css > min.css
```

Input
```css
ul,
ol,
li,
h1,
h2,
h3,
h4,
h5,
h6,
pre,
form,
p,
blockquote,
th,
td {
    margin: 0;
    padding: 0;
}

ul,
ol,
dl {
    line-height: 1.25;
    margin-bottom: 1rem;
    list-style-position: outside;
}

li {
    font-size: inherit;
}

ul {
    list-style-type: disc;
    margin-left: 1.25rem;
}

ol {
    margin-left: 1.25rem;
}

ul ul, ol ul, ul ol, ol ol {
    margin-left: 1.25rem;
    margin-bottom: 0;
    list-style-type: inherit;
}
```

Output (with source formatting applied)
```css
ol, ul {
    margin-left: 1.25rem;
}

blockquote, form, h1, h2, h3, h4, h5, h6, li, ol, p, pre, td, th, ul {
    margin: 0;
    padding: 0;
}

dl, ol, ul {
    line-height: 1.25;
    margin-bottom: 1rem;
    list-style-position: outside;
}

li {
    font-size: inherit;
}

ul {
    list-style-type: disc;
}

ol ol, ol ul, ul ol, ul ul {
    margin-left: 1.25rem;
    margin-bottom: 0;
    list-style-type: inherit;
}
```

**Comment by jakubpawlowicz** [comment]
Thanks @fergaldoyle, it's a bug indeed. I'll post a fix soon, please use `--skip-restructuring` in the meanwhile.

**Comment by jakubpawlowicz** [comment]
Already fixed in 3.4.9 by f3c32e67988c2d0bc05dbf5ad9c296b6bea5eced.
